docs/spec_draft.md

```diff
@@ -1529,6 +1530,64 @@ operations correspond to [stablehlo.minimum](#stablehlominimum) and
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.collective_permute
+
+### Semantics
+
+Within each process group in the StableHLO grid, sends and receives data between
+processes in the group.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+  * `channel_id <= 0`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0`,
+    `cross_partition(replica_groups)`.
+
+Afterwards, `result@target` is given by
+ * `operand@process_groups[i][0]`, if there exist an `i` such that  `process_groups[i][1] = target`.
+ * `result@target = T, otherwise, # where T is a zero valued tensor constant with type(T) = type(operand)`, otherwise.
+
+### Inputs
+
+| Name                  | Type                                         |
+|-----------------------|----------------------------------------------|
+| `operand`             | tensor of any supported type                 |
+| `source_target_pairs` | 2-dimensional tensor constant of type `si64` |
+| `channel_id`          | constant of type `si64`                      |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) dim(`source_target_pairs`, 1) $=$ 2.
+  * (C2) All members of { `source_target_pairs`[i][0] : i $\in$ [0, dim(`source_target_pairs`, 0)) } are unique.
+  * (C3) All members of { `source_target_pairs`[i][1] : i $\in$ [0, dim(`source_target_pairs`, 0)) } are unique.
```

Comment:
How about "All values in `source_target_pairs[:, 0]` are unique" and the same for C3?